SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -0,0 +1,351 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+ 
+#include "Utils/EntityFactory.h"
+ 
+#include "Engine.h"
+ 
+#include "EngineClasses/SpatialActorChannel.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/SpatialRPCService.h"
+#include "Schema/AlwaysRelevant.h"
+#include "Schema/AuthorityIntent.h"
+#include "Schema/Heartbeat.h"
+#include "Schema/ClientRPCEndpointLegacy.h"
+#include "Schema/ServerRPCEndpointLegacy.h"
+#include "Schema/RPCPayload.h"
+#include "Schema/Singleton.h"
+#include "Schema/SpawnData.h"
+#include "Utils/ComponentFactory.h"
+#include "Utils/InterestFactory.h"
+#include "Utils/SpatialActorUtils.h"
+ 
+namespace SpatialGDK
+{
+ 
+EntityFactory::EntityFactory(USpatialNetDriver* InNetDriver, SpatialRPCService* InRPCService)
+	: NetDriver(InNetDriver)
+	, PackageMap(InNetDriver->PackageMap)
```

Comment:
I would prefer if PackageMap and ClassInfoManager were passed in separately rather than being taken of the Netdriver

Reply:
As @MatthewSandfordImprobable says - we've agreed to pass all arguments explicitly, not get variables from the NetDriver internally. Unfortunately, in this case we'd still have to pass the NetDriver itself, since there's a dependency on it.

Reply:
Makes sense; I'll make this change (as well as anything that trickles down from that).

Reply:
Addressed in 45c78cb, thanks for calling this out!